File: tools/project-memory-context/tests/fixtures/sample.go
package main

import "fmt"

type Animal struct {
    Name string
}

type Speaker interface {
    Speak() string
}

func (a Animal) Speak() string {
    return a.Name
}

func NewAnimal(name string) Animal {
    return Animal{Name: name}
}

func privateHelper(x int) int {
    return x * 2
}

File: tools/project-memory-context/tests/fixtures/sample.rs
pub struct Animal {
    pub name: String,
}

pub enum Status {
    Active,
    Inactive,
}

pub trait Speaker {
    fn speak(&self) -> String;
}

impl Animal {
    pub fn new(name: &str) -> Self {
        Animal { name: name.to_string() }
    }
}

pub fn standalone(x: i32) -> i32 {
    x * 2
}

fn private_helper() {}
